contracts/SignerBondsUniswapV2.sol

```diff
@@ -0,0 +1,260 @@
+        swapDeadline = _swapDeadline;
+    }
+
+    /// @notice Swaps signer bonds on Uniswap v2 exchange.
+    /// @dev Swaps the given ETH amount for the target token using the Uniswap
+    ///      decentralized exchange. The maximum ETH amount is capped by the
+    ///      contract balance. Some governance parameters are applied on the
+    ///      transaction. The swap's price impact must fit within the
+    ///      maximum allowed price impact and the transaction is constrained
+    ///      with the slippage tolerance and deadline. Acquired target tokens
+    ///      are sent to the recipient address set during contract construction.
+    /// @param amount Amount to swap.
+    function swapSignerBondsOnUniswapV2(uint256 amount) external {
+        require(amount > 0, "Amount must be greater than 0");
+        require(amount <= address(this).balance, "Amount exceeds balance");
+
+        // Setup the swap path. WETH must be the first component.
+        address[] memory path = new address[](2);
+        path[0] = uniswapRouter.WETH();
+        path[1] = targetToken;
+
+        // Calculate the maximum output token amount basing on pair reserves.
+        // This value will be used as the minimum amount of output tokens that
+        // must be received for the transaction not to revert.
+        // This value includes liquidity fee equal to 0.3%.
+        uint256 amountOutMin = uniswapRouter.getAmountsOut(amount, path)[1];
+
+        require(
+            isAllowedPriceImpact(amountOutMin),
+            "Price impact exceeds allowed limit"
+        );
+
+        // Include slippage tolerance into the minimum amount of output tokens.
+        amountOutMin = amountOutMin
+            .mul(BASIS_POINTS_DIVISOR.sub(slippageTolerance))
+            .div(BASIS_POINTS_DIVISOR);
+
+        // TODO: Do not send acquired tokens to the asset pool in case an
+        //       open auction exists. In that case, additional tokens will
+        //       cause the auction to offer more of the coverage pool value for
+        //       the same price. This is unreasonable in terms of economy.
+        //       Figure out how to address that problem.
+
+        // slither-disable-next-line arbitrary-send,reentrancy-events
+        uint256[] memory amounts =
+            uniswapRouter.swapExactETHForTokens{value: amount}(
+                amountOutMin,
+                path,
+                targetTokenRecipient,
+                /* solhint-disable-next-line not-rely-on-time */
+                block.timestamp.add(swapDeadline)
+            );
+
+        emit UniswapV2SwapExecuted(amounts);
+    }
+
+    /// @notice Checks the price impact of buying a given amount of tokens
+    ///         against the maximum allowed price impact limit.
+    /// @param amount Amount of tokens.
+    /// @return True if the price impact is allowed, false otherwise.
+    function isAllowedPriceImpact(uint256 amount) internal view returns (bool) {
```

Comment:
Wouldn't it be better to keep it public so that one can do a kind of a pre-flight before calling `swapSignerBondsOnUniswapV2`?

Reply:
Why not. See https://github.com/keep-network/coverage-pools/pull/66/commits/1788cc3fd025d7fb6e0bb34e55376bc7e21c01de